More tests for _

danvk opened the pull request:
A bug in the declaration of `_.zip`, caught by a test!

samwgoldman commented:
I think the old type of `_.zip` was correct. `_.zip([1], ['a']) === [[1, 'a']]`, which is `[number,string][]`.

danvk commented:
Ah, you're right! I changed the declaration back and updated the tests.

It would be nice to turn these into executable tests (e.g. using Mocha), though if FlowTyped gets to be anywhere near the size of DefinitelyTyped, that could have problems scaling.
